### PR TITLE
Added example for extracting sensor data from email body using regex.

### PR DESCRIPTION
**Pull request by scstraus**
## Proposed change

This is a good working example of how to turn an email body into useful sensors which I think would save a lot of people time in building the same. I believe this is the second most common use case for this component other than the first example I gave.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.
- [X] Added an example to existing documentation

## Additional information

- Link to parent pull request in the codebase:
- This PR fixes or closes issue:

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_integrations/imap_email_content.markdown [hunk 1]
```diff
@@ -98,3 +98,47 @@ sensor:
 {% endraw %}
 
 The same template structure can scan the date, body or sender for matching text before setting the state of the sensor.
+
+## Example - extracting formatted text from an email using template sensors
+
+This example shows how to extract numbers or other formatted data from an email to change the value of a template sensor to a value extracted from the email. In this example, we will be extracting energy use, cost, and billed amount from an email (from Georgia Power) and putting it into sensor values using a template sensor which runs against our IMAP email sensor already set up. An sample of the body of the email used is below:
+{% raw %}
```
Comment (frenck) [inline source/_integrations/imap_email_content.markdown]: Raw is not needed here.

File: source/_integrations/imap_email_content.markdown [hunk 2]
```diff
@@ -98,3 +98,47 @@ sensor:
 {% endraw %}
 
 The same template structure can scan the date, body or sender for matching text before setting the state of the sensor.
+
+## Example - extracting formatted text from an email using template sensors
+
+This example shows how to extract numbers or other formatted data from an email to change the value of a template sensor to a value extracted from the email. In this example, we will be extracting energy use, cost, and billed amount from an email (from Georgia Power) and putting it into sensor values using a template sensor which runs against our IMAP email sensor already set up. An sample of the body of the email used is below:
```
Comment (frenck) [inline source/_integrations/imap_email_content.markdown]:
```suggestion
This example shows how to extract numbers or other formatted data from an email to change the value of a template sensor to a value extracted from the email. In this example, we will be extracting energy use, cost, and billed amount from an email (from Georgia Power) and putting it into sensor values using a template sensor that runs against our IMAP email sensor already set up. A sample of the body of the email used is below:
```

File: source/_integrations/imap_email_content.markdown [hunk 3]
```diff
@@ -98,3 +98,47 @@ sensor:
...
+    billing_cycle_total:
+      friendly_name: Billing Cycle Total
+      unit_of_measurement: $
+      value_template: >
+        {{ state_attr('sensor.energy_email', 'body')
+          |regex_findall_index("\ days:\* \$([0-9.]+)") }}
+```
+{% endraw %}
+
+By making small changes to the regular expressions defined above, a similar structure can parse other types of data out of the body of other emails.
```
Comment (frenck) [inline source/_integrations/imap_email_content.markdown]:
```suggestion
By making small changes to the regular expressions defined above, a similar structure can parse other types of data out of the body of other emails.
```

**Review by frenck — CHANGES_REQUESTED** [review]
Thanks for the PR @scstraus!

I left some small comments/suggestions. Could you take a look? Thanks! 👍

**Comment by scstraus** [comment]
All changes accepted. Was too busy double checking syntax and forgot to check grammar ;-).

**Comment by frenck** [comment]
One comment is unresolved + the build is still failing.
Please make sure you address all of that, thanks! 👍

**Comment by scstraus** [comment]
Okay think I got the last one, sorry.

**Comment by klaasnicolaas** [comment]
Fixed the build errors for you 😄

**Review by klaasnicolaas — APPROVED** [review]

**Review by frenck — APPROVED** [review]
Yep! Thanks guys! 👍

**Review by scstraus — COMMENTED** [review]
Thanks guys, I suppose it's obvious by now that I am hopeless when it comes to github.. But I know how hard these examples were to come by so I wanted to make sure they were preserved for others.

**Pull request merged**